### PR TITLE
audio: delete switch component

### PR DESCRIPTION
**Pull request by cujomalainey**
mostly not implemented and hasn't been meaningfully touched in > 3 yr.

### CODE REVIEW

**Review by btian1 — COMMENTED** [review]
it's an empty component, delete it make sense.

**Review by btian1 — APPROVED** [review]

File: src/audio/CMakeLists.txt [hunk 1]
```diff
@@ -57,11 +57,6 @@ if((NOT CONFIG_LIBRARY) OR CONFIG_LIBRARY_STATIC)
 	if(CONFIG_COMP_MUX)
 		add_subdirectory(mux)
 	endif()
-	if(CONFIG_COMP_SWITCH)
```
Comment (lyakh) [inline src/audio/CMakeLists.txt:60]: but we keep `CONFIG_COMP_SWITCH` ?

**Review by lyakh — APPROVED** [review]

**Review by dbaluta — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]
Ack never used - @cujomalainey can be removed from kernel too.

**Comment by dbaluta** [comment]
@cujomalainey there are some more references to `CONFIG_COMP_SWITCH`. Can we remove those also?
```
src/arch/xtensa/configs/mt8186_defconfig:CONFIG_COMP_SWITCH=n
src/arch/xtensa/configs/mt8188_defconfig:CONFIG_COMP_SWITCH=n
src/arch/xtensa/configs/mt8195_defconfig:CONFIG_COMP_SWITCH=n
src/arch/xtensa/configs/override/tigerlake_chrome_google_aec.config:CONFIG_COMP_SWITCH=n
src/arch/xtensa/configs/override/tigerlake_chrome_waves_google_aec.config:CONFIG_COMP_SWITCH=n
src/arch/xtensa/configs/rembrandt_defconfig:CONFIG_COMP_SWITCH=n
src/audio/CMakeLists.txt:       if(CONFIG_COMP_SWITCH)
zephyr/CMakeLists.txt:zephyr_library_sources_ifdef(CONFIG_COMP_SWITCH
```

**Comment by cujomalainey** [comment]
> @cujomalainey there are some more references to `CONFIG_COMP_SWITCH`. Can we remove those also?

Nice catch, thanks I was grepping at src/audio. Fixed now

**Comment by kv2019i** [comment]
Looks good to go. Let's wait for the CI results. It seems we have an issue with SOF and Zephyr mainline again. Won't block this PR, but something we need to address.

**Comment by marc-hb** [comment]
> It seems we have an issue with SOF and Zephyr mainline again.

It's not Zephyr actually, just SOF. Tentative fix submitted in #7945

**Pull request merged**